src/utils/sort-versions.ts

```diff
@@ -2,28 +2,54 @@ import * as semver from 'semver';
 
 import { RunnableVersion } from '../interfaces';
 
-function electronSemVerCompare(a: semver.SemVer, b: semver.SemVer) {
-  const l = a.compareMain(b);
-  if (l) return l;
-  // Electron's approach is nightly -> other prerelease tags -> stable,
-  // so force `nightly` to sort before other prerelease tags.
-  const [prea] = a.prerelease;
-  const [preb] = b.prerelease;
-  if (prea === 'nightly' && preb !== 'nightly') return -1;
-  if (prea !== 'nightly' && preb === 'nightly') return 1;
-  return a.comparePre(b);
-}
+const preTags = ['nightly', 'alpha', 'beta'];
 
-function compare(a: semver.SemVer | string, b: semver.SemVer | string) {
-  // non-semver goes first, sorted lexicographically
-  const astr = typeof a === 'string';
-  const bstr = typeof b === 'string';
-  if (astr && bstr) return (a as string).localeCompare(b as string);
-  if (astr) return -1;
-  if (bstr) return 1;
+/**
+ * Sorts prerelease tags such that nightly -> alpha -> beta.
+ *
+ * @param a Prerelease tag data for the old version.
+ * @param b Prerelease tag data for the new version.
+ * @returns 0 | 1 | -1
+ */
+const preCompare = (
+  a: readonly (string | number)[],
+  b: readonly (string | number)[],
+) => {
+  const one = preTags.indexOf(a[0] as string);
+  const two = preTags.indexOf(b[0] as string);
+  if (one === two) {
+    // Whether the prerelease tag number is the same
+    // e.g. alpha.1 & alpha.1.
+    if (a[1] === b[1]) return 0;
+    return a[1] > b[1] ? 1 : -1;
+  }
 
-  // then semver, sorted newest-to-oldest
-  return -electronSemVerCompare(a as semver.SemVer, b as semver.SemVer);
+  return one > two ? 1 : -1;
+};
+
+/**
+ * Custom semver comparator which takes into account Electron's prerelease
+ * tag hierarchy.
+ *
+ * Sorts in ascending order when passed to Array.sort().
+ *
+ * @param a The old Electron version.
+ * @param b The new Electron version.
+ * @returns 0 | 1 | -1
+ */
+export function semverCompare(a: string, b: string) {
+  const hasPre = (v: string) => preTags.some((tag) => v.includes(tag));
+
+  // Check that major.minor.patch are the same for a and b.
+  if (semver.coerce(a)?.raw === semver.coerce(b)?.raw) {
+    if (hasPre(a) && hasPre(b)) {
+      const { prerelease: aPre } = semver.parse(a) as semver.SemVer;
+      const { prerelease: bPre } = semver.parse(b) as semver.SemVer;
+      return preCompare(aPre, bPre);
+    }
```

Comment:
- Is coerce() needed here, or will parse() do just as well? I don't think I understand the need for coercion here.  I *think* we already ensure semver correctness when Fiddle loads up the version array to begin with.

- the values of coerce() are discarded, only to turn around and re-parse the same string all over again 2-3 lines below

- And all of this is done in the compare func, meaning that we're back to doing something like 4 * O(n^2) parses again. This discards the "parse in O(N), not O(N^2)" behavior that was in the implementation of sortVersions()

Reply:
this is a workaround for checking that `a.major === b.major && a.minor === b.minor && a.patch === b.patch` - i'll rework it.